jbehave-core/src/main/java/org/jbehave/core/reporters/ConcurrentStoryReporter.java

```diff
@@ -92,7 +92,7 @@
         }
     }
 
-    private List<DelayedMethod> delayedMethods = new ArrayList<>();
+    private List<DelayedMethod> delayedMethods = Collections.synchronizedList(new ArrayList<DelayedMethod>());
```

Comment:
What's the reason of this change? It may significantly decrease performance

Reply:
If it's not synchronized the number of beforeStep invocations and sucessfull/failed invocation is not equal